windows-gaming-driver/src/controller.rs

```diff
@@ -121,11 +88,10 @@ impl Controller {
 
         // send GA all hotkeys we want to register
         for (i, hotkey) in self.machine_config.hotkeys.clone().iter().enumerate() {
-            let mut buf = Vec::new();
-            buf.write_u32::<LittleEndian>(i as u32).unwrap();
-            buf.write_u32::<LittleEndian>(hotkey.key.len() as u32).unwrap();
-            buf.extend(hotkey.key.bytes());
-            self.write_ga_buf(GaCmd::RegisterHotKey, &buf);
+            self.write_ga(GaCmd::RegisterHotKey {
+                id: i as u32,
+                key: hotkey.key.clone(),
```

Comment:
Can't - we clone it once for the controller, but `ga_hello()` can run multiple times.